src/coreclr/jit/lsra.cpp

```diff
@@ -418,6 +418,8 @@ void LinearScan::updateRegsFreeBusyState(RefPosition& refPosition,
 
 regMaskTP LinearScan::internalFloatRegCandidates()
 {
+    needFloatingRegisters = true;
```

Comment:
Does this also cover `maskRegisters`? Should there be a separate concept or should they be tied together?

Reply:
I think they should better be tied together for now. We can revisit it in future.

Reply:
Should we maybe call this `needNonGeneralRegisters` or `needNonIntegerRegisters` to possibly be more inclusive and less confusing as to what exactly it covers?

Reply:
Sure, will change it.